clustering/preprocess.py

```diff
@@ -19,8 +19,28 @@ def filter_restaurants(businesses, max_restaurants=None):
     (determined by looking at the review data).
     """
 
-    # TODO: filter the input DataFrame
-    return businesses
+    cols = ['business_id', 'is_open', 'review_count', 'categories']
+    businesses = businesses[cols]
+
+    businesses = businesses[businesses['review_count'] > 10]
+    businesses = businesses[businesses['is_open'] != 0]
+
+    valid_b = businesses.as_matrix()
+    b = np.empty_like(businesses['business_id'])
+
+    for i in range(businesses.shape[0]):
+        if "Food" or "Restaurant" or "food" or "restaurant" in valid_b[i, 3]:
+            b[i] = valid_b[i, 0]
+
+    businesses = businesses[businesses['business_id'].isin(b)]
```

Comment:
Minor nit, but I like using different variable names instead of overwriting the same one over and over again, so here, since you'll have filtered by restaurants, you could create a new `restaurants` DataFrame view, which improves the readability of the code.